brackend/schemagen.py

```diff
@@ -0,0 +1,165 @@
+                    winners[i][j][1][1] = 1
+                    winners[i][j][1][2] = len(bracket["losers"][1]["seeds"])-j-1
+                    winners[i][j][1][3] = 0
+                    cur = winners[i][j][1]
+                    bracket[cur[0]][cur[1]]["seeds"][cur[2]]["teams"][cur[3]]["name"] = [i,j]
+                elif i > 1:
+                    midindex = len(bracket["losers"][(2 * (i+1)) - 3]["seeds"])//2
+                    winners[i][j][1][1] = (2 * (i+1)) - 3
+                    if i % 2 == 1:
+                        if j < midindex:
+                            winners[i][j][1][2] = midindex + j
+                        else:
+                            winners[i][j][1][2] = j - midindex
+                    else:
+                        winners[i][j][1][2] = j
+                    winners[i][j][1][3] = 0
+                    cur = winners[i][j][1]
+                    bracket[cur[0]][cur[1]]["seeds"][cur[2]]["teams"][cur[3]]["name"] = [i,j]
+    #print(bracket)
+    return winners
+
+# This takes in a bracket made by makeBracketFromEntrants and the number of people in that bracket and returns a schema.
+# A schema is a set of instructions for where to send the winners and losers of each match in the bracket.
+def makeSchemaJSON(bracket2, e):
+    bracket = copy.deepcopy(bracket2)
+    winners = [[[["winners",j+1,None,None],["losers",None,None,None]] for k in range(len(bracket["winners"][j]["seeds"]))] for j in range(len(bracket["winners"]))]
+    losers = [[[["losers",j+1,None,None],"OUT"] for k in range(len(bracket["losers"][j]["seeds"]))] for j in range(len(bracket["losers"]))]
+    # Whether or not there are the same number of matches in LR1 and LR2
+    # affects how the losers are handled.
+    same_next = 0
+    if len(bracket["losers"][0]["seeds"]) == len(bracket["losers"][1]["seeds"]):
+        #print(len(bracket["losers"][1]["seeds"]))
+        same_next = 1
+    for i in range(len(winners)-1):
+        for j in range(len(winners[i])):
+            #print(i, j, bracket["winners"][i], winners[i])
+            if bracket["winners"][i]["seeds"][j]["teams"] != [{'name': '', 'game': -1}, {'name': '', 'game': -1}]:
+                #print(bracket["winners"][i]["seeds"][j]["teams"])
+                if len(winners[i+1]) == len(winners[i]) and i == 0:
+                    winners[i][j][0][2] = j
+                    winners[i][j][0][3] = 1
+                    #print(winners[i][j], "HERE")
+                    cur = winners[i][j][0]
+                    #print(bracket[cur[0]][cur[1]]["seeds"][cur[2]]["teams"][cur[3]])
+                    bracket[cur[0]][cur[1]]["seeds"][cur[2]]["teams"][cur[3]]["name"] = [i,j]
+                else:
+                    winners[i][j][0][2] = j//2
+                    winners[i][j][0][3] = j%2
+                    #print(winners[i][j], "HERE")
+                    cur = winners[i][j][0]
+                    bracket[cur[0]][cur[1]]["seeds"][cur[2]]["teams"][cur[3]]["name"] = [i,j]
```

Comment:
really long indices like this are very unclear. Try to use named intermediate variables to explain what is going on more clearly.